### PR TITLE
typing improvements on level and fill_value in Series.eq

### PR DESCRIPTION
**Pull request by color455nm**
- [x] item 6 and 7 in [issue#11](https://github.com/noatamir/pyladies-aug23-sprint/issues/11)
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

### CODE REVIEW

**Comment by color455nm** [comment]
pyladies-aug23-sprint : @phofl @noatamir

File: pandas/core/series.py [hunk 1]
```diff
@@ -5630,7 +5630,7 @@ def dropna(
 
     def to_timestamp(
         self,
-        freq=None,
+        freq: str | None = None,
```
Comment (twoertwein) [inline pandas/core/series.py:5633]: I believe you can use `pandas._typing.Frequency` as this ends up in `PeriodArray.to_timestamp` which supports that.

**Review by twoertwein — COMMENTED** [review]
One small comment, otherwise looks good to me!

File: pandas/core/series.py (hunk 1 above)
Comment (color455nm) [reply pandas/core/series.py]: sounds good, will update

**Review by twoertwein — APPROVED** [review]

**Comment by twoertwein** [comment]
Thanks @color455nm !

**Pull request merged**